internal/operator-controller/rukpak/bundle/config.go

```diff
@@ -0,0 +1,108 @@
+package bundle
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/yaml"
+
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+)
+
+type Config struct {
+	WatchNamespace string `json:"watchNamespace"`
+}
+
+// UnmarshallConfig returns a deserialized and validated *bundle.Config based on bytes and validated
+// against rv1 and the desired install namespaces. It will error if:
+// - rv is nil
+// - bytes is not a valid YAML/JSON object
+// - bytes is a valid YAML/JSON object but does not follow the registry+v1 schema
+// if bytes is nil a nil bundle.Config is returned
+func UnmarshallConfig(bytes []byte, rv1 *RegistryV1, installNamespace string) (*Config, error) {
+	if bytes == nil {
+		return nil, nil
+	}
+	if rv1 == nil {
+		return nil, errors.New("bundle is nil")
+	}
+
+	bundleConfig := &Config{}
+	if err := yaml.UnmarshalStrict(bytes, bundleConfig); err != nil {
+		return nil, fmt.Errorf("error unmarshalling registry+v1 configuration: %w", formatUnmarshallError(err))
+	}
+
+	if err := validateConfig(bundleConfig, rv1, installNamespace); err != nil {
+		return nil, fmt.Errorf("error unmarshalling registry+v1 configuration: %w", err)
+	}
+
+	return bundleConfig, nil
+}
+
+func validateConfig(config *Config, rv1 *RegistryV1, installNamespace string) error {
+	// no config, no problem
+	if config == nil {
+		return nil
+	}
+
+	// collect bundle install modes
+	installModeSet := sets.New(rv1.CSV.Spec.InstallModes...)
+
+	// only accept a non-empty value for watchNamespace if the bundle configuration accepts the watchNamespace config
+	if config.WatchNamespace != "" && !hasWatchNamespaceAsConfig(installModeSet) {
+		return errors.New(`unknown field "watchNamespace"`)
```

Comment:
given that `WatchNamespace` is not empty, I believe that the error should not be `unknown field...`?